### PR TITLE
Master chrome headless devtools moc

### PR DESCRIPTION
**Pull request by antonylesuisse**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -38,3 +38,7 @@ setup/win32/static/postgresql*.exe
 /man/
 /share/
 /src/
+
```
Comment (antonylesuisse) [inline .gitignore]: remve

File: odoo/tests/common.py [hunk 2]
```diff
@@ -594,6 +601,385 @@ def phantom_js(self, url_path, code, ready="window", login=None, timeout=60, **k
...
+
+
+    def take_screenshot(self, prefix='failed'):
+        ss_id = self._websocket_send('Page.captureScreenshot')
+        _logger.info('Asked for screenshot (id: %s)' % ss_id)
+        res = self._websocket_wait_id(ss_id)
+        base_png = res.get('result', {}).get('data')
+        decoded = base64.decodebytes(bytes(base_png.encode('utf-8')))
+        timestamp = datetime.now().strftime('%Y%m%d_%H%M%S_%f')
+        outpath = os.path.join(self.screenshots_dir, '%s_%s.png' % (prefix,timestamp))
```
Comment (antonylesuisse) [inline odoo/tests/common.py]: '%s_failed_%Y%m%d_%H%M%S_%f_%d' % (os.path.splitext('logfile.log')[0], 1)

File: addons/point_of_sale/tests/test_frontend.py [hunk 3]
```diff
@@ -9,7 +9,9 @@
 
 
 class TestUi(odoo.tests.HttpCase):
-    def test_01_pos_basic_order(self):
+
+    def setUp(self):
+        super(TestUi, self).setUp()
```
Comment (antonylesuisse) [inline addons/point_of_sale/tests/test_frontend.py]: remove

File: addons/point_of_sale/tests/test_point_of_sale_ui.py [hunk 4]
```diff
@@ -7,4 +7,4 @@
 class TestUi(odoo.tests.HttpCase):
 
     def test_01_point_of_sale_tour(self):
-        self.phantom_js("/web", "odoo.__DEBUG__.services['web_tour.tour'].run('point_of_sale_tour')", "odoo.__DEBUG__.services['web_tour.tour'].tours.point_of_sale_tour.ready", login="admin")
+        self.phantom_js("/web", "odoo.__DEBUG__.services['web_tour.tour'].run('point_of_sale_tour')", "odoo.__DEBUG__.services['web_tour.tour'].tours.point_of_sale_tour.ready", timeout=120, login="admin")
```
Comment (antonylesuisse) [inline addons/point_of_sale/tests/test_point_of_sale_ui.py]: remove

File: odoo/addons/test_assetsbundle/tests/test_assetsbundle.py [hunk 5]
```diff
@@ -362,7 +362,8 @@ def test_01_js_interpretation(self):
         self.phantom_js(
             "/test_assetsbundle/js",
             "a + b + c === 6 ? console.log('ok') : console.log('error')",
-            login="admin"
+            login="admin",
+            wait_page_stop=True
```
Comment (antonylesuisse) [inline odoo/addons/test_assetsbundle/tests/test_assetsbundle.py]: remove

File: odoo/tests/common.py (hunk 2 above)
Comment (d-fence) [reply odoo/tests/common.py]: On the runbot, the stderr and stdout are redirected to a file, there is no --logfile argument

https://github.com/odoo/runbot/blob/e815571d14163c8af6c19637c244ff022abaa883/runbot/models/build.py#L716-L718

**Comment by Yajo** [comment]
Google Chrome is closed source; can this tool be configured to use the open source Chromium instead?

**Comment by Yajo** [comment]
Please see #26464

**Pull request merged**